src/slot_provider/slot.rs

```diff
@@ -0,0 +1,66 @@
+use std::{fmt::Display, ops::Add};
+
+use chrono::{DateTime, Duration, Utc};
+use eyre::Result;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    pub static ref BEACON_GENESIS: DateTime<Utc> = "2020-12-01T12:00:23Z".parse().unwrap();
+}
+
+/// A slot number on the beacon chain.
+/// Started at 0 at 2020-12-01T12:00:23Z (genesis). Slots follow unix timestamps meaning most slots
+/// are 12 seconds long, but some are 13 or 11 seconds long. We use u32 to store. Enough for 12
+/// seconds * 2^62 = ~1634 years.
+#[derive(Clone, Copy, Debug)]
+pub struct Slot(pub u32);
```

Comment:
If a struct only has a single field you can make it a unit struct. One big benefit is if you ever serialize it you can tell serde to treat it as if it was just a u32. Same for deserializing. Unfortunately ssz throws a wrench into this. Haven't tried to make it work but that may be but a small step away.

Reply:
Makes sense.
This change might be the reason for the incompatibility with the current server version though. (just a hint)